module/apmhttp/clienttrace.go

```diff
@@ -0,0 +1,97 @@
+package apmhttp
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http/httptrace"
+
+	"go.elastic.co/apm"
+)
+
+// WithClientTrace returns a ClientOption for
+// tracing events within HTTP client requests.
+func WithClientTrace() ClientOption {
+	return func(rt *roundTripper) {
+		rt.requestTracer = &requestTracer{}
+	}
+}
+
+type requestTracer struct {
+	tx *apm.Transaction
+
+	DNS,
+	Connect,
+	TLS,
+	Server,
+	Transfer *apm.Span
+}
+
+func (r *requestTracer) start(ctx context.Context) context.Context {
+	if r == nil {
```

Comment:
Can you please move the nil checks outside? That's how most of the Go agent code is written, and I would prefer to keep it consistent.